### PR TITLE
#541 (1) - Refactor properties.

### PR DESCRIPTION
**Pull request by treshugart**
This PR removes all of the redundant options and code since we're making Skate a little bit more opinionated. This also simplifies the property lifecycles since it now syncs attribute to property from the `attributeChangedCallback` in v0 and whatever they're calling it now in v1.

### CODE REVIEW

File: src/lifecycle/created.js [hunk 1]
```diff
@@ -27,9 +23,23 @@ function ensurePropertyDefinitions (elem, propertyFunctions) {
   }, {});
 }
 
+function iniitaliseProperties (elem, propertyDefinitions) {
```
Comment (joscha) [inline src/lifecycle/created.js]: just a small thing, but there us a common typo :)

File: src/lifecycle/properties-init.js [hunk 2]
```diff
@@ -4,147 +4,90 @@ import data from '../util/data';
...
-function getData (elem, name) {
-  return data(elem, `api/property/${name}`);
+function getDefaultValue (elem, name, opts) {
+  return typeof opts.default === 'function' ? opts.default(elem, { name }) : opts.default;
 }
 
-function getDataForAttribute (elem, name) {
-  return getData(elem, getData(elem, name).linkedProperty);
+function getInitialValue (elem, name, opts) {
+  typeof opts.initial === 'function' ? opts.initial(elem, { name }) : opts.initial;
```
Comment (nadiam84) [inline src/lifecycle/properties-init.js]: no "return"?

Comment (joscha) [inline src/lifecycle/properties-init.js]: 👍  I think there might be a test missing as well then :)

Comment (treshugart) [inline src/lifecycle/properties-init.js]: Added some tests. It required a tiny bit of refactoring in the test to accommodate `initial` tests, but happy they got written. Totally my oversight they didn't get in initially (no pun intended).

File: src/lifecycle/created.js (hunk 1 above)
Comment (treshugart) [inline src/lifecycle/created.js]: Fixed.

**Pull request merged**